test/Microsoft.AspNetCore.Mvc.Core.Test/Routing/UrlHelperTest.cs

```diff
@@ -971,6 +971,46 @@ public void Action_RouteValueInvalidation_DoesNotAffectActionAndController_Actio
             Assert.Equal(expected, url);
         }
 
+        [Fact]
+        public void GetUrlHelper_ReturnsSameInstance_IfAlreadyPresent()
+        {
+            // Arrange
+            var expectedUrlHelper = CreateUrlHelper();
+            var httpContext = new Mock<HttpContext>();
+            var mockItems = new Dictionary<object, object>
+            {
+                { typeof(IUrlHelper), expectedUrlHelper }
+            };
```

Comment:
Also test the case where the dictionary has a null/non-IUrlHelper since you added support for it.